src/storage/iceberg_delete.cpp

```diff
@@ -0,0 +1,358 @@
+	auto &local_state = input.local_state.Cast<IcebergDeleteLocalState>();
+
+	auto &file_name_vector = chunk.data[row_id_indexes[0]];
+	auto &file_row_number = chunk.data[row_id_indexes[1]];
+
+	UnifiedVectorFormat row_data;
+	file_row_number.ToUnifiedFormat(chunk.size(), row_data);
+	auto file_row_data = UnifiedVectorFormat::GetData<int64_t>(row_data);
+
+	UnifiedVectorFormat file_name_vdata;
+	file_name_vector.ToUnifiedFormat(chunk.size(), file_name_vdata);
+	for (idx_t i = 0; i < chunk.size(); i++) {
+		auto row_idx = row_data.sel->get_index(i);
+		auto file_name_idx = file_name_vdata.sel->get_index(i);
+		if (!file_name_vdata.validity.RowIsValid(file_name_idx)) {
+			throw InternalException("Filename cannot be NULL!");
+		}
+		auto file_name_data = UnifiedVectorFormat::GetData<string_t>(file_name_vdata);
+		auto file_name = file_name_data[file_name_idx].GetString();
+
+		if (local_state.current_file_name.empty() || local_state.current_file_name != file_name) {
+			// file has changed - flush
+			global_state.Flush(local_state);
+			local_state.current_file_name = file_name;
+			// insert the file name for the file if it has not yet been inserted
+			auto entry = local_state.filenames.find(file_name);
+			if (entry == local_state.filenames.end()) {
+				local_state.filenames.emplace(file_name);
+			}
+		}
+		auto row_number = file_row_data[row_idx];
+		local_state.file_row_numbers.push_back(row_number);
+	}
+	return SinkResultType::NEED_MORE_INPUT;
+}
+
+//===--------------------------------------------------------------------===//
+// Combine
+//===--------------------------------------------------------------------===//
+SinkCombineResultType IcebergDelete::Combine(ExecutionContext &context, OperatorSinkCombineInput &input) const {
+	auto &global_state = input.global_state.Cast<IcebergDeleteGlobalState>();
+	auto &local_state = input.local_state.Cast<IcebergDeleteLocalState>();
+	global_state.FinalFlush(local_state);
+	return SinkCombineResultType::FINISHED;
+}
+
+static optional_ptr<CopyFunctionCatalogEntry> TryGetCopyFunction(DatabaseInstance &db, const string &name) {
+	D_ASSERT(!name.empty());
+	auto &system_catalog = Catalog::GetSystemCatalog(db);
+	auto data = CatalogTransaction::GetSystemTransaction(db);
+	auto &schema = system_catalog.GetSchema(data, DEFAULT_SCHEMA);
+	return schema.GetEntry(data, CatalogType::COPY_FUNCTION_ENTRY, name)->Cast<CopyFunctionCatalogEntry>();
+}
+
+//===--------------------------------------------------------------------===//
+// Finalize
+//===--------------------------------------------------------------------===//
+void IcebergDelete::FlushDelete(IRCTransaction &transaction, ClientContext &context,
+                                IcebergDeleteGlobalState &global_state, const string &filename,
+                                vector<idx_t> &deleted_rows) const {
+	// find the matching data file for the deletion
+	auto data_file_info = delete_map->GetExtendedFileInfo(filename);
```

Comment:
`data_file_info` is unused here
This is the only place that `GetExtendedFileInfo` is used, so I think that method can also be removed

Which also means that `AddExtendedFileInfo` is not necessary ?
And by association the `file_map`